test/extended/operators/olm.go

```diff
@@ -168,6 +168,28 @@ var _ = g.Describe("[sig-operator] an end user can use OLM", func() {
 	g.It("can subscribe to the operator", func() {
 		g.By("Cluster-admin user subscribe the operator resource")
 
+		// skip test if redhat-operators is not present or disabled
+		sources, err := oc.AsAdmin().Run("get").Args("operatorhub/cluster", "-o", "jsonpath='{.spec.sources}'").Output()
+		o.Expect(err).NotTo(o.HaveOccurred())
+		type Source struct {
+			Name     string `json:"name"`
+			Disabled bool   `json:"disabled"`
+		}
+		parsed := []Source{}
+		err = json.Unmarshal([]byte(sources), &parsed)
+		o.Expect(err).NotTo(o.HaveOccurred())
+
+		hasRedHatOperators := false
+		for _, source := range parsed {
+			if source.Name == "redhat-operators" && source.Disabled == false {
```

Comment:
Don't you want to check the inverse of this? if `source.Disabled == true` then the redhat operator catalog is not enabled on the cluster.

Reply:
Updated the following line to read `hasRedHatOperators = true` when "redhat-operators" is present and not disabled